{{cookiecutter.repository_name}}/operations/jupyter.py

```diff
@@ -1,28 +1,58 @@
+import shutil
 import argparse
 import subprocess
-import os
+import webbrowser
 from pathlib import Path
+from notebook.services.contents.filemanager import FileContentsManager
 
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--port", type=int, default=8888)
-    args = parser.parse_args()
+def jupyter(notebook_name, port, clean, interactive):
+    original_path = Path("notebooks") / notebook_name
+    notebook_path = Path(notebook_name) if clean else original_path
+    if original_path.exists() and not notebook_path.exists():
+        shutil.copy(original_path, notebook_path)
+    if not notebook_path.exists():
+        FileContentsManager().new(path=notebook_path.as_posix())
 
-    working_dir = f"{os.environ['GUILD_HOME']}/runs/{os.environ['RUN_ID']}"
-    command = f"jupyter notebook --port={args.port}"
+    if not clean:
+        package_name = "{{cookiecutter.package_name}}"
+        Path(f"sourcecode-symlink/{package_name}").rename(package_name)
 
-    if "PYTHONSTARTUP" in os.environ and Path(os.environ["PYTHONSTARTUP"]).exists():
-        python_startup_script = (
-            Path(os.environ["PYTHONSTARTUP"]).open("r").read() + "\n"
-        )
+    if interactive:
+        subprocess.run(["jupyter", "notebook", notebook_path.as_posix()])
     else:
-        python_startup_script = ""
+        subprocess.run(
+            [
+                "jupyter",
+                "nbconvert",
+                "--to",
+                "notebook",
+                "--inplace",
+                "--execute",
+                notebook_path.as_posix(),
+            ]
+        )
 
-    Path("change-dir.py").write_text(
-        f"{python_startup_script}import os\nos.chdir('{working_dir}')"
-    )
+    subprocess.run(["jupyter", "nbconvert", "--to", "html", notebook_path.as_posix()])
+    if not interactive:
+        webbrowser.open(notebook_path.with_suffix(".html").as_posix())
 
-    os.environ["PYTHONSTARTUP"] = f"{os.getcwd()}/change-dir.py"
 
-    subprocess.run(command, shell=True)
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--notebook", type=str, required=True)
+    parser.add_argument("--port", type=int, default=8888)
+    parser.add_argument(
```

Comment:
Can use `action=store_true` to avoid having to set a value :slightly_smiling_face:

Reply:
This might not work with guild, im not sure